_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,374 @@
+---
+layout: post
+title:  "Reassigning partitions in Apache Kafka Cluster"
+date: 2021-09-23
+author: shubham_rawat
+---
+
+As Apache Kafka users, we commonly deal with brokers and the partitions. 
+Sometimes you might want to move the partitions from old brokers to the new brokers or maybe you want to change the ordering of the partition assignment list.
+Cruise Control can be used to tackle these issues easily but for any reason if you don't want to use Cruise Control and do the things manually then you can use the Kafka partition reassignment tool.
+Here is a guide on how you can use it.
+
+<!--more-->
+
+##  Kafka reassignment partition tool
+
+The `bin/kafka-reassign-partitions.sh` tool allows you to reassign partitions to different brokers.
+
+While using the tool, you have to provide it with either of these two JSON files: `topics.json` and `reassignment.json`.
+Wondering what these two JSON files really do? Let's talk a bit about them:
+
+- The `topic.json` basically consists of the topics that we want to reassign or move. Based on this JSON file, our tool will generate a proposal `reassignment.json` that we can use directly or modify further.
+
+- The `reassignment.json` file is a configuration file that is used during the partition reassignment process. The reassignment partition tool will generate a proposal `reassignment.json` file based on a `topics.json` file. You can change the `reassignment.json` file as per your requirement and use it.
+
+## Why use the Kafka reassignment partition tool?
+
+The Kafka reassignment partition tool can help you address a variety of use cases.
+
+Some of these are listed here:
+
+- You can reassign partitions between the brokers. For example, it can be used when you want to scale down the number of brokers in your cluster.
+   You can assign partitions from the broker to be scaled down to other brokers which will handle these partitions now.
+
+- You can increase the number of partitions / replicas which can help with increasing the throughput of topics.
+
+## Partition Reassignment Throttle
+Reassigning partitions between brokers often leads to additional interbroker network traffic, in addition to the normal traffic required for replication. To avoid overloading the cluster, it is recommended to always set a throttle rate to limit the bandwidth used by the reassignment. 
```

Comment:
```suggestion
Reassigning partitions between brokers often leads to additional inter-broker network traffic, in addition to the normal traffic required for replication. 
To avoid overloading the cluster, it is recommended to always set a throttle rate to limit the bandwidth used by the reassignment. 
```